Sage/carbon
# refactor(inline-inputs): convert files to typescript FE-4859

### Proposed behaviour


Refactors `InlineInputs`, `InputBehaviour` and `CarbonProvider` into TypeScript
Updates type definition in `label.d.ts` to make `htmlFor` optional

### Current behaviour


`InlineInputs`, `InputBehaviour` and `CarbonProvider` are JavaScript
Type definition in `label.d.ts` to make `htmlFor` is required

### Checklist




#### QA


### Additional context



### Testing instructions


Test `InlineInputs` for regressions and test inputs (RadioButton and Checkbox etc) to check that updates to InputBehaviour has not caused regression. Check themes work in storybook to confirm no regressions following updates to `CarbonProvider`.

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
